packages/widgetbook/lib/src/state/widgetbook_state.dart

```diff
@@ -144,12 +144,18 @@ class WidgetbookState extends ChangeNotifier {
   /// rest of fields that stay unchanged during runtime.
   @internal
   void updateFromRouteConfig(AppRouteConfig routeConfig) {
-    path = routeConfig.path;
-    previewMode = routeConfig.previewMode;
-    queryParams = {
-      // Copy from UnmodifiableMap
-      ...routeConfig.queryParameters
-    };
+    if (routeConfig.path != null && catalog.hasPath(routeConfig.path!)) {
+      path = routeConfig.path;
+      previewMode = routeConfig.previewMode;
+      queryParams = {
+        // Copy from UnmodifiableMap
+        ...routeConfig.queryParameters
+      };
+    } else {
+      path = null;
+      previewMode = false;
+      queryParams = {};
+    }
```

Comment:
Why do we need this part?

Reply:
Without this, if the path doesn't match a route the app gets stuck. You can select use cases but won't load them properly (knobs missing was the big one, URL not updating).

Can occur with someone editing the URL or sending a link to a use case that has moved or been deleted.

Reply:
We can split this fix to another PR, to discuss it further there.
PRs should be small scoped and fixing/adding one thing at a time.

Reply:
I'd recommend keeping the entire change set as this is to prevent undefined behaviour the change in `widgetbook.dart` would otherwise allow.